src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -544,24 +544,24 @@ private static bool TryParseVersion(string version, ref VersionResult result)
 
             var versionSansLabel = (dashIndex < 0) ? version : version.Substring(0, dashIndex);
             string[] parsedComponents = versionSansLabel.Split(Utils.Separators.Dot);
-            if (parsedComponents.Length != 3)
+            if (parsedComponents.Length == 0 || parsedComponents.Length > 3)
```

Comment:
I believe that's the ETS .Length returning 1 where we treat scalars as arrays and add both .Length and .Count properties.  If you do [string]::Empty.Split('.')[0].Length, you get 0 as expected